packages/analytics-js-plugins/src/deviceModeDestinations/utils.ts

```diff
@@ -22,65 +57,155 @@ const isIntegrationSDKEvaluated = (pluginName: string, modName: string, logger?:
+        callback?: ApiCallback,
+      ) => analytics.alias(aliasArgumentsToCallOptions(to, from, options, callback)),
+      group: (
+        groupId: string | number | Nullable<ApiObject> | ApiCallback,
+        traits?: Nullable<ApiOptions> | Nullable<ApiObject> | ApiCallback,
+        options?: Nullable<ApiOptions> | ApiCallback,
+        callback?: ApiCallback,
+      ) => analytics.group(groupArgumentsToCallOptions(groupId, traits, options, callback)),
+      getAnonymousId: () => analytics.getAnonymousId(),
+      getUserId: () => analytics.getUserId(),
+      getUserTraits: () => analytics.getUserTraits(),
+      getGroupId: () => analytics.getGroupId(),
+      getGroupTraits: () => analytics.getGroupTraits(),
+      getSessionId: () => analytics.getSessionId(),
+    } as DeviceModeDestinationsAnalyticsInstance,
+    {
+      areTransformationsConnected: dest.areTransformationsConnected,
+      destinationId: dest.id,
     },
-    intg.destinationInfo,
   );
 };
 
-const isInitialized = (instance: any, time = 0) =>
-  new Promise(resolve => {
-    if (instance.isLoaded()) {
-      console.log('instance.isLoaded', instance);
+const isDestinationReady = (dest: Destination, logger?: ILogger, time = 0) =>
+  // eslint-disable-next-line compat/compat
+  new Promise((resolve, reject) => {
+    const instance = dest.instance as DeviceModeDestination;
+    if (instance.isLoaded() && (!instance.isReady || instance.isReady())) {
       resolve(this);
     } else if (time >= INITIALIZED_CHECK_TIMEOUT) {
-      throw Error('instance.init timeout expired');
+      reject(new Error(`Destination "${dest.userFriendlyId}" ready check timed out`));
     } else {
-      pause(LOAD_CHECK_POLL_INTERVAL)
+      wait(LOAD_CHECK_POLL_INTERVAL)
         .then(() =>
-          isInitialized(instance, time + LOAD_CHECK_POLL_INTERVAL)
+          isDestinationReady(dest, logger, time + LOAD_CHECK_POLL_INTERVAL)
             .then(resolve)
-            .catch(() => {}),
+            .catch(e => reject(e)),
         )
-        .catch(() => {});
+        .catch(e => reject(e));
     }
   });
 
-export { isIntegrationSDKEvaluated, pause, createIntegrationInstance, isInitialized };
+/**
+ * Converts the common names of the destinations to their display names
+ * @param intgOptions Load or API integration options
+ */
+const normalizeIntegrationOptions = (intgOptions?: IntegrationOpts): IntegrationOpts => {
+  const normalizedIntegrationOptions: IntegrationOpts = {};
+  if (intgOptions) {
+    Object.keys(intgOptions).forEach(key => {
+      const destOpts = clone(intgOptions[key]);
+      if (key === 'All') {
+        normalizedIntegrationOptions[key] = Boolean(destOpts);
+      } else {
+        const displayName = destCNamesToDispNamesMap[key];
+        if (displayName) {
+          normalizedIntegrationOptions[displayName] = destOpts;
+        } else {
+          normalizedIntegrationOptions[key] = destOpts;
+        }
```

Comment:
Can you explain the scenario when this part of the code(line 164-166) will be useful?

Reply:
For cloud mode destinations, there won't be any entry in the cnames map. Hence, we use that field as it is.